cpp/cmake/thirdparty/CUDF_GetcuCollections.cmake

```diff
@@ -0,0 +1,38 @@
+#=============================================================================
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+function(find_and_configure_cucollections)
+
+    if(TARGET cuCollections::cuco)
+        return()
+    endif()
+
+    # Find or install cuCollections
+    CPMFindPackage(NAME   cuco
+        GITHUB_REPOSITORY NVIDIA/cuCollections
+        GIT_TAG           dev
```

Comment:
I expect we should use an explicit SHA1 so we can control when breaking changes are introduced.